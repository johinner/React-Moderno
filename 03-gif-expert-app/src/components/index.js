export * from './AddCategory';
export * from './GifGrid';
export * from './GitfItem';
