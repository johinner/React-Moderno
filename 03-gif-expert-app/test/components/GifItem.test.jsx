import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GitfItem";

describe("Prueba en componente <GifItem />", () => {
    const title = "prueba Testing";
    const url = "https://pruebatesting.jpg/";

    //fotocopia del componente en html
    test("debe de hacer match con el snapshot", () => {
        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test('debe de mostrar la imagen con el URL y el ALT indicado', () => {
        render(<GifItem title={title} url={url} />);
        //screen.debug();
        const {src, alt} = screen.getByRole('img')
        expect(alt).toBe(title);
        expect(src).toBe(url);
    })
});
