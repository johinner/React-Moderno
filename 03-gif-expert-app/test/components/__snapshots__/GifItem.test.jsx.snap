// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en componente <GifItem /> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="prueba Testing"
      src="https://pruebatesting.jpg/"
    />
    <p>
      prueba Testing
    </p>
  </div>
</div>
`;
